ohara-agent/src/main/scala/com/island/ohara/agent/BrokerCollie.scala

```
@@ -79,6 +65,19 @@ object BrokerCollie {
     private[this] var exporterPort: Int = BrokerApi.EXPORTER_PORT_DEFAULT
     private[this] var jmxPort: Int = BrokerApi.JMX_PORT_DEFAULT
 
+    override def copy(clusterInfo: ClusterInfo): ClusterCreator.this.type = clusterInfo match {
+      case bk: BrokerClusterInfo =>
+        super.copy(clusterInfo)
+        zookeeperClusterName(bk.zookeeperClusterName)
+        clientPort(bk.clientPort)
+        exporterPort(bk.exporterPort)
+        jmxPort(bk.jmxPort)
+        this
+      case _ =>
+        super.copy(clusterInfo)
+        this
+    }
+
```

Since this is in a subclass, should we throw an exception instead of use super.copy in the `others` case?

---

sure